File: src/cpp/consoleUtil.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
#include <iomanip>
#include "results.cpp"

using namespace std;

class ConsoleUtil
{
  public:
    static string Prompt(string question)
    {
        cout << question << endl;
        string result;
        cin >> result;
        return result;
    }

    static void DisplayMatrixResult(results::MatrixResult result)
    {
        vector<vector<int>> matrix = result.Matrix;
        int length = max(to_string(result.Side * result.Side).length() + 1, static_cast<unsigned int>(2));

        for (int rowIndex = 0; rowIndex < result.Side; rowIndex++)
        {
            for (int columnIndex = 0; columnIndex < result.Side; columnIndex++)
                cout << setw(length) << matrix[rowIndex][columnIndex];
            cout << endl;
        }
        cout << "Sum: " << result.Sum;
    }
};

File: src/rs/src/console_util.rs
use std::cmp;

use crate::results;

pub fn prompt(question: &str) -> String {
    println!("{}", question);
    let mut input = String::new();
    std::io::stdin().read_line(&mut input).unwrap();
    return input;
}

pub fn display_matrix_result(result: results::MatrixData) -> () {
    let matrix = result.matrix;
    let side = result.side;
    let length = cmp::max((side * side).to_string().len() + 1, 2);
    for row_index in 0..side {
        for column_index in 0..side {
            print!("{:1$}", matrix[row_index][column_index], length);
        }
        println!();
    }
    println!("Sum: {}", result.sum);
}
